### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by lucifercr07**
Unit tests being added for the following files:
- cookie.js
- header.js
- request-auth.js
- item.js
- content-info.js

### CODE REVIEW

File: test/unit/cookie.test.js [hunk 1]
```diff
@@ -194,4 +194,135 @@ describe('Cookie', function () {
...
+        });
+    });
+
+    describe('unparseSingle', function () {
+        it('should return only the value in case of no cookie name specified', function () {
+            var rawCookie = {
+                    expires: 1502442248,
+                    hostOnly: false,
+                    httpOnly: false,
+                    key: 'foo',
```
Comment (coditva) [inline test/unit/cookie.test.js:213]: We have specified a `key` here. Shouldn't it be used for `name`? 🤔

Comment (coditva) [inline test/unit/cookie.test.js:213]: I think it's worth adding a test to check that when `name` is not specified, `key` is used instead.

File: test/unit/cookie.test.js [hunk 2]
```diff
@@ -194,4 +194,135 @@ describe('Cookie', function () {
...
+                unparse;
+            cookieArray.push(rawCookie1);
+            cookieArray.push(rawCookie2);
+
+            unparse = Cookie.unparse(cookieArray);
+            expect(unparse).equals('testCookie1=fooTest1; testCookie2=fooTest2');
+        });
+    });
+
+    describe('stringify', function () {
```
Comment (coditva) [inline test/unit/cookie.test.js:268]: Shouldn't we add tests for `stringify` similar to all tests for `toString`?

File: test/unit/cookie.test.js [hunk 3]
```diff
@@ -194,4 +194,135 @@ describe('Cookie', function () {
...
+            var cookie = new Cookie({
+                key: 'foo',
+                value: 'fooTest',
+                maxAge: 1502442248
+            });
+            expect(cookie.toString())
+                .equals('foo=fooTest; Max-Age=1502442248');
+        });
+    });
 });
```
Comment (coditva) [inline test/unit/cookie.test.js:328]: Should we also add tests for cases when each of the options are invalid? `null`, `object` instead of `string`, etc.

File: test/unit/cookie.test.js [hunk 4]
```diff
@@ -194,4 +194,135 @@ describe('Cookie', function () {
...
+                    hostOnly: false,
+                    httpOnly: false,
+                    key: 'foo',
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'bar'
+                },
+                unparsedSingle = Cookie.unparseSingle(rawCookie);
+            expect(unparsedSingle).equals('bar');
```
Comment (coditva) [inline test/unit/cookie.test.js:220]: Let's change `.equals` to `.to.equal` because it is more readable and forms a sentence:
> expect **actual value** to equal **expected value**

File: test/unit/header.test.js [hunk 5]
```diff
@@ -70,6 +70,15 @@ describe('Header', function () {
                 value: 0
             });
         });
+
+        it('should create the header with system options', function () {
+            var header = Header.create({ key: 'name', value: 'value', system: true });
+            expect(header.toJSON()).to.eql({
+                key: 'name',
+                value: 'value',
+                system: true
```
Comment (coditva) [inline test/unit/header.test.js:79]: What happens when `system` is not boolean?

File: test/unit/header.test.js [hunk 6]
```diff
@@ -70,6 +70,15 @@ describe('Header', function () {
                 value: 0
             });
         });
```
Comment (coditva) [inline test/unit/header.test.js:72]: Let's add tests for invalid values of `key` as well.

File: test/unit/header.test.js [hunk 7]
```diff
@@ -229,4 +238,13 @@ describe('Header', function () {
...
     });
+
+    describe('toString', function () {
+        it('should convert the header to a single header string', function () {
+            // eslint-disable-next-line max-len
+            var rawHeader = 'Content-Type: application/json\nAuthorization: Hawk id="dh37fgj492je", ts="1448549987", nonce="eOJZCd", mac="O2TFlvAlMvKVSKOzc6XkfU6+5285k5p3m5dAjxumo2k="',
+                newHeader = new Header(rawHeader);
+            expect(newHeader.toString()).equals(rawHeader);
+        });
+    });
```
Comment (coditva) [inline test/unit/header.test.js:249]: Let's also add a test where we create a new header by passing an object to `new Header()` and check the `toString` output.

File: test/unit/item.test.js [hunk 8]
```diff
@@ -466,6 +466,27 @@ describe('Item', function () {
...
             expect(item.toJSON()).to.have.property('protocolProfileBehavior').that.is.empty;
         });
+
+        it('should not unset protocolProfileBehavior from an Item incase key is not of type string', function () {
+            var item = new Item({
+                protocolProfileBehavior: { keyName: null }
+            });
+
+            item.unsetProtocolProfileBehavior(null);
+            expect(item.toJSON()).to.have.property('protocolProfileBehavior').that.is.not.empty;
```
Comment (coditva) [inline test/unit/item.test.js:476]: Let's also assert on the value of `item.protocolProfileBehavior`

File: test/unit/item.test.js [hunk 9]
```diff
@@ -466,6 +466,27 @@ describe('Item', function () {
             item.unsetProtocolProfileBehavior('keyName');
             expect(item.toJSON()).to.have.property('protocolProfileBehavior').that.is.empty;
         });
+
+        it('should not unset protocolProfileBehavior from an Item incase key is not of type string', function () {
+            var item = new Item({
+                protocolProfileBehavior: { keyName: null }
```
Comment (coditva) [inline test/unit/item.test.js:472]: Why is the value `null`? It makes the test a little confusing.

File: test/unit/request-auth.test.js [hunk 10]
```diff
@@ -4,123 +4,185 @@ var expect = require('chai').expect,
...
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        auth.use('basic');
-        expect(auth.type).to.equal('basic');
-        expect(auth.current()).to.eql({
-            username: 'u', password: 'p'
+            auth.use('noauth', {
```
Comment (coditva) [inline test/unit/request-auth.test.js:87]: Why are we setting the auth options again in the `.use` method? The test name does not say anything about it.
It is making the `expect` a little confusing.

If we're testing that `.use` can update the options, I think we can break this test into two.

File: test/unit/request-auth.test.js [hunk 11]
```diff
@@ -4,123 +4,185 @@ var expect = require('chai').expect,
...
+                noauth: {
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        expect(auth.current()).to.eql({});
+            auth.update(null);
+            auth.update({});
+            expect(auth.parameters().toObject()).to.eql({
```
Comment (coditva) [inline test/unit/request-auth.test.js:130]: It would make more sense to either break this test into multiple tests for each invalid value, or add expect after each `.update`.

We should also check for other invalid values like `string`, `numbers`, `boolean`, etc.

**Review by coditva — CHANGES_REQUESTED** [review]

**Review by coditva — CHANGES_REQUESTED** [review]
- [x] Also need to update `CHANGELOG.yaml`

File: test/unit/request-auth.test.js (hunk 11 above)
Comment (lucifercr07) [reply test/unit/request-auth.test.js:143]: Fixed

File: test/unit/request-auth.test.js (hunk 10 above)
Comment (lucifercr07) [reply test/unit/request-auth.test.js:87]: It's the latter one, have updated the test description to depict that we are updating the auth via options.

File: test/unit/cookie.test.js (hunk 1 above)
Comment (coditva) [reply test/unit/cookie.test.js]: I think this might be a bug? We allow using `key` in place of `name` when creating a cookie. Ref: [/lib/collection/cookie.js#L125](https://github.com/postmanlabs/postman-collection/blob/16cc6f821a2af99d979477f9d97432b1d5362090/lib/collection/cookie.js#L125).

@codenirvana views?

File: test/unit/request-auth.test.js (hunk 10 above)
Comment (coditva) [reply test/unit/request-auth.test.js:87]: I think it's better for readability that we pass just `{ foo1: 'bar1' }` rather than the whole thing again.
It also makes the usage of the function more clear.

**Review by coditva — CHANGES_REQUESTED** [review]

File: test/unit/cookie.test.js (hunk 2 above)
Comment (coditva) [reply test/unit/cookie.test.js:290]: 🎉

File: test/unit/cookie.test.js [hunk 12]
```diff
@@ -287,19 +288,63 @@ describe('Cookie', function () {
...
 
     describe('stringify', function () {
+        it('should return single Set-Cookie header string with key value', function () {
+            var cookie = new Cookie({
+                name: 'foo',
+                value: 'fooTest'
+            });
+
+            expect(Cookie.stringify(cookie))
+                .equals('foo=fooTest');
```
Comment (coditva) [inline test/unit/cookie.test.js:298]: Nitpick: `.to.equals` allows better readability. And we can keep it on the same line.

It's your choice to change it in this PR. But we can keep in mind for later PRs.

**Review by coditva — DISMISSED** [review]
LGTM! 🚀

File: test/unit/cookie.test.js [hunk 13]
```diff
@@ -46,6 +46,34 @@ describe('Cookie', function () {
...
+                    hostOnly: false,
+                    httpOnly: false,
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'fooTest'
+                }),
+                unparsedSingle = Cookie.unparseSingle(testCookie);
+
+            expect(unparsedSingle).to.equal(testCookie.name + '=' + testCookie.value);
```
Comment (codenirvana) [inline test/unit/cookie.test.js:74]: Assert the exact string value instead of concatenating.

File: test/unit/cookie.test.js [hunk 14]
```diff
@@ -193,5 +221,178 @@ describe('Cookie', function () {
...
+                    expires: 1502442248,
+                    hostOnly: false,
+                    httpOnly: false,
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'fooTest'
+                },
+                unparsedSingle = Cookie.unparseSingle(new Cookie(rawCookie));
+            expect(unparsedSingle).to.equal(rawCookie.name + '=' + rawCookie.value);
```
Comment (codenirvana) [inline test/unit/cookie.test.js:246]: Assert the exact string.

File: test/unit/cookie.test.js [hunk 15]
```diff
@@ -193,5 +221,178 @@ describe('Cookie', function () {
...
+                    value: 'fooTest1'
+                },
+                rawCookie2 = {
+                    name: 'testCookie2',
+                    domain: '.httpbin.org',
+                    expires: 1502442248,
+                    path: '/',
+                    value: 'fooTest2'
+                },
+                cookieArray = [],
```
Comment (codenirvana) [inline test/unit/cookie.test.js:280]:
```suggestion
                cookieArray = [rawCookie1, rawCookie2],
```

File: test/unit/cookie.test.js [hunk 16]
```diff
@@ -193,5 +221,178 @@ describe('Cookie', function () {
...
+
+    describe('stringify', function () {
+        it('should return single Set-Cookie header string with key value', function () {
+            var cookie = new Cookie({
+                name: 'foo',
+                value: 'fooTest'
+            });
+
+            expect(Cookie.stringify(cookie))
+                .equals('foo=fooTest');
```
Comment (codenirvana) [inline test/unit/cookie.test.js:298]:
```suggestion
            expect(Cookie.stringify(cookie)).to.equals('foo=fooTest');
```

File: test/unit/cookie.test.js [hunk 17]
```diff
@@ -193,5 +221,178 @@ describe('Cookie', function () {
...
+        });
+
+        it('should return single Set-Cookie header string with expires', function () {
+            var cookie = new Cookie({
+                name: 'foo',
+                value: 'fooTest',
+                expires: '14 Jun 2017 00:00:00 PDT'
+            });
+
+            expect(Cookie.stringify(cookie))
```
Comment (codenirvana) [inline test/unit/cookie.test.js:308]: `.to. equals`

File: test/unit/cookie.test.js [hunk 18]
```diff
@@ -193,5 +221,178 @@ describe('Cookie', function () {
...
+        it('should return single Set-Cookie header string with secure and httpOnly set to true', function () {
+            var cookie = new Cookie({
+                name: 'foo',
+                value: 'fooTest',
+                httpOnly: true,
+                secure: true
+            });
+
+            expect(Cookie.stringify(cookie))
+                .equals('foo=fooTest; Secure; HttpOnly');
```
Comment (codenirvana) [inline test/unit/cookie.test.js:321]: Split into multiple lines only if it exceeds 120 chars limit.

File: test/unit/content-info.test.js [hunk 19]
```diff
@@ -174,6 +174,28 @@ describe('contentInfo module', function () {
             });
     });
 
+    it('Should return default file name if charset specified in content-disposition header is invalid', function () {
```
Comment (codenirvana) [inline test/unit/content-info.test.js:177]:
```suggestion
    it('should return default file name if charset specified in content-disposition header is invalid', function () {
```

File: test/unit/content-info.test.js [hunk 20]
```diff
@@ -174,6 +174,28 @@ describe('contentInfo module', function () {
...
+            {
+                key: 'Content-Disposition',
+                value: 'attachment;filename*=invalid-charset\'\'%E5%93%8D%E5%BA%94.json'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON()
+        });
```
Comment (codenirvana) [inline test/unit/content-info.test.js:188]:
```suggestion
        var response = new Response({
            header: [{
                key: 'Content-Disposition',
                value: 'attachment;filename*=invalid-charset\'\'%E5%93%8D%E5%BA%94.json'
            }, {
                key: 'Content-Type',
                value: 'application/json'
            }],
            stream: Buffer.from('a test json').toJSON()
        });
```

fix formatting.

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: test/unit/content-info.test.js (hunk 19 above)
Comment (lucifercr07) [reply test/unit/content-info.test.js]: Fixed.

File: test/unit/content-info.test.js (hunk 20 above)
Comment (lucifercr07) [reply test/unit/content-info.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 13 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 14 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 15 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 12 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 16 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 17 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 18 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 1 above)
Comment (codenirvana) [reply test/unit/cookie.test.js]: Yes, this looks like a mistake but it's fine for now. I will take a look at this.

**Review by codenirvana — APPROVED** [review]
